orca-core/src/main/java/com/netflix/spinnaker/orca/pipeline/util/ArtifactResolver.java

```diff
@@ -196,33 +187,55 @@ public void resolveArtifacts(@Nonnull Map pipeline) {
 
   public Artifact resolveSingleArtifact(ExpectedArtifact expectedArtifact, List<Artifact> possibleMatches) {
     List<Artifact> matches = possibleMatches
-      .stream()
-      .filter(expectedArtifact::matches)
-      .collect(toList());
+        .stream()
+        .filter(expectedArtifact::matches)
+        .collect(toList());
     switch (matches.size()) {
       case 0:
         return null;
-      case 1:
-        return matches.get(0);
       default:
-        throw new IllegalStateException(format("Expected artifact %s matches multiple artifacts %s", expectedArtifact, matches));
```

Comment:
Curious why we are now allowing multiple artifacts to match and picking the first, rather than expecting exactly one to match as before

Reply:
This code was originally written when expected artifacts only lived in triggers - and we generally treat artifacts at the start of execution as unordered. But with this change, allowing expected artifact in the middle of an execution it's desirable to always bind the most recent artifact.

Thinking about it I'll add a flag to make sure still fail triggers with ambiguous matches